### ISSUE TITLE
Run pipeline: allow executing a single/specific pipeline if more than one pipeline definitions exist

### ISSUE THREAD

**Issue by pat-s** [woodpecker-ci/woodpecker#2763]
### Clear and concise description of the problem

The "Run pipeline" button in the UI, which is triggered for `"event: manual"`, currently executes all existing pipeline definitions under `.woodpecker/`.

### Suggested solution

The button could allow invoking specific pipelines only instead of starting all existing ones.

### Validations

- [X] Checked that the feature isn't part of the `next` version already [https://woodpecker-ci.org/faq#which-version-of-woodpecker-should-i-use]
- [X] Read the [Contributing Guidelines](https://github.com/woodpecker-ci/woodpecker/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't already an [issue](https://github.com/woodpecker-ci/woodpecker/issues) that request the same feature to avoid creating a duplicate.

**Comment by lukasz-zaroda** [comment]
This feature is very important for my use case. Basically, what I need to do is to have a different pipeline that would build the database for my app, and a different pipeline that would deploy the app by using this database. These steps actually don't depend on each other. The database is a derivative of another live database. The app can be deployed without building the database (in this case, the last one built will be used). The idea being that the database would be periodically built by using cron, and the app deployed only when code changes. Currently, it seems to be not possible to include these semi-dependent pipelines in a single repository. The only workaround that comes to my mind is to create a dedicated repository for the "database building" pipeline, which would work, but seems to be unnecessary. I would love to be able to handle everything related to the project, in this project's repository, and not creating utility repositories just to run other project related pipelines independently.

**Comment by pat-s** [comment]
> The idea being that the database would be periodically built by using cron, and the app deployed only when code changes

If you set different `when` triggers in the individual pipeline YAMLS, this should work. It becomes more complicated if you want to trigger individual pipelines manually.

**Comment by lukasz-zaroda** [comment]
@pat-s I see. That's one solution, however, the manual pipeline triggering is an ultimate fallback if anything goes wrong for whatever reason. Not having it is making this setup very uncomfortable. Let's say the requirement for database building changes, at one point in a way, that cannot be represented by cron. Let's say it should be a manual triggering based on some external event. And suddenly, you need to rework your pipelines. I think the individual manual triggering is the most basic requirement because it can handle the most situations. If something is difficult to automate for whatever reason, you can do manual triggering as a workaround before you'll figure this out. Not having this option is frankly bizarre, but It's awesome that it's getting recognized.

**Comment by cabezayunke** [comment]
any updates on this? is it planned? I am in a similar situation as lukasz...

**Comment by cabezayunke** [comment]
FYI, I am going to try and implement this, I'll keep you posted here

**Comment by cabezayunke** [comment]
I am working on this now... I would like to double check a specific behaviour about this:

when a workflow has a `depends_on` it shouldn't be allowed to run on its own right? should go hand by hand with its dependent workflow? so server should validate dependencies before running individual workflows and in the UI, help with this by automatically checking/unchecking dependencies and showing a toast to make it clear to the user what is happening... 

For example here, selecting `deploy`, will select `build` and `test` automatically:
<img width="1347" height="761" alt="Image" src="https://github.com/user-attachments/assets/61ecb9fd-f1c0-415e-bcf7-834e957dd549" />

And unchecking the `build`, will uncheck the `deploy` too, since it cant deploy without the build:
<img width="1338" height="757" alt="Image" src="https://github.com/user-attachments/assets/917b2daf-4b99-47be-98ed-71e39d211a12" />

I hope this makes sense, any feedback is welcome!

**Comment by lukasz-zaroda** [comment]
I think autoselecting dependencies will be confusing as long as the dependency tree is not visible here. Could we write the dependencies next to workflow names? So, for example:

- build (depends_on: lint)
- deploy (depends_on: build, test)
- lint
- notify
- test

And then if someone selects `deploy` and runs this workflow, the the dependencies will run first. Then it could be a single-value select. But if you would still prefer the multi-value one, then at least add information about dependencies to the list, if that's possible.

**Comment by qwerty287** [comment]
I would say it's ok to just ignore the depends_on in case the dependencies are not selected. This should be clear to the user that if they have a "hard" dependency (i.e. using artifacts from another workflow) in their configs they also need to select the dependencies, but I think many people are also using depends_on just to "order" the workflows.

**Comment by cabezayunke** [comment]
> I would say it's ok to just ignore the depends_on in case the dependencies are not selected. This should be clear to the user that if they have a "hard" dependency (i.e. using artifacts from another workflow) in their configs they also need to select the dependencies, but I think many people are also using depends_on just to "order" the workflows.

so for the scenario where it is a real/hard dependency, the run would just fail? 
or should the dependencies still run automatically so everything works?

based on this, the single value check (ignoring dependencies) should be fine, it the run fails because dependencies are not selected, it is on the user, right?
I think this might be acceptable together with the info about dependencies... agreed?

**Comment by qwerty287** [comment]
Yes, I would say the users will know which pipelines they can run standalone and which ones not. If you run a workflow without its hard dependencies, it's fine that it fails then.

**Comment by cabezayunke** [comment]
PR open: https://github.com/woodpecker-ci/woodpecker/pull/7159
